File: src/main/java/com/novafurniture/NovaFurniture/service/ReviewService.java
package com.novafurniture.NovaFurniture.service;

import com.novafurniture.NovaFurniture.dto.request.ReviewRequest;
import com.novafurniture.NovaFurniture.dto.request.UpdateReviewRequest;
import com.novafurniture.NovaFurniture.dto.response.PageResponse;
import com.novafurniture.NovaFurniture.dto.response.ReviewResponse;

public interface ReviewService {
    ReviewResponse createReview(Long userId, ReviewRequest request);
    ReviewResponse updateReview(Long userId, Long reviewId, UpdateReviewRequest request);
    void deleteReview(Long userId, Long reviewId);
    PageResponse<ReviewResponse> getReviewsByProduct(Long productId, int page, int size);
    PageResponse<ReviewResponse> getMyReviews(Long userId, int page, int size);
    Double getAverageRating(Long productId);
}

File: fe/src/services/reviewService.js
import axiosClient from './axiosClient';

const reviewService = {
    // GET /reviews/product/{productId}
    getReviewsByProduct: (productId, params = {}) =>
        axiosClient.get(`/reviews/product/${productId}`, { params }),

    // GET /reviews/product/{productId}/rating
    getRatingByProduct: (productId) =>
        axiosClient.get(`/reviews/product/${productId}/rating`),

    // POST /reviews
    createReview: (data) =>
        axiosClient.post('/reviews', data),

    // PUT /reviews/{reviewId}
    updateReview: (reviewId, data) =>
        axiosClient.put(`/reviews/${reviewId}`, data),

    // DELETE /reviews/{reviewId}
    deleteReview: (reviewId) =>
        axiosClient.delete(`/reviews/${reviewId}`)
};

export default reviewService;